### PR TITLE
Support for selecting an elliptic curve for key exchange

### PR DESCRIPTION
**Pull request by Mike-Ki-ASD**
#### Description

Added option to specify an elliptic curve for key exchange between mitmproxy <-> server

#### Checklist

 - [x] I have updated tests where applicable.
 - [x] I have added an entry to the CHANGELOG.

### CODE REVIEW

File: mitmproxy/net/tls.py [hunk 1]
```diff
@@ -106,6 +107,13 @@ def _create_ssl_context(
     # Options
     context.set_options(DEFAULT_OPTIONS)
 
+    # ECDHE for Key exchange
+    if ctx.options.curve is not None:
```
Comment (mhils) [inline mitmproxy/net/tls.py:111]: The return values of methods in `net/tls.py` should depend on their arguments only, not on externally-defined variables/options. This needs to be accessed in tlsconfig.py and passed in as an argument.

File: mitmproxy/options.py [hunk 2]
```diff
@@ -74,6 +74,13 @@ def __init__(self, **kwargs) -> None:
             None,
             "Set supported ciphers for mitmproxy <-> server connections using OpenSSL syntax.",
         )
+        self.add_option(
+            "curve",
+            Optional[str],
+            None,
+            "Specify an elliptic curve for ECDHE key exchange between mitmproxy <-> server connections using OpenSSL "
+            "syntax."
```
Comment (mhils) [inline mitmproxy/options.py:82]: Add a curve example here?

**Review by mhils — COMMENTED** [review]
Thanks for the PR! :)

File: mitmproxy/options.py (hunk 2 above)
Comment (Mike-Ki-ASD) [reply mitmproxy/options.py]: I added 3 example curve, however I was not sure whether I should specify the command `openssl ecparam -list_curve`. This command lists the supported curves, at least for Linux.

**Review by mhils — APPROVED** [review]
Thanks again! 🍰

**Pull request merged**